src/companies/services/stock/material_type_creator.py

```diff
@@ -0,0 +1,28 @@
+import logging
+from typing import Callable
+
+from rest_framework.serializers import BaseSerializer
+
+from app.services import BaseService
+from companies.api.serializers import MaterialTypeSerializer
+from companies.models import MaterialType
+
+
+class MaterialTypeCreator(BaseService):
+    def __init__(self, serializer: BaseSerializer[MaterialTypeSerializer]) -> None:
+        self.serializer = serializer
+        self.logger = logging.getLogger(__name__)
```

Comment:
This have to be initiated on the module level

logger = logging.getLogger(__name__)

```suggestion


class MaterialTypeCreator(BaseService):
    def __init__(self, serializer: BaseSerializer[MaterialTypeSerializer]) -> None:
        self.serializer = serializer
```